examples/org.eclipse.cdt.lsp.examples.preferences/src/org/eclipse/cdt/lsp/examples/preferences/MyClangdOptionsDefaults.java

```diff
@@ -27,19 +26,9 @@ public class MyClangdOptionsDefaults extends ConfigurationMetadataBase implement
 
 	@Override
 	protected List<PreferenceMetadata<?>> definePreferences() {
-		List<PreferenceMetadata<?>> defined = new ArrayList<>();
-		defined.add(clangdPath);
-		defined.add(useTidy);
-		defined.add(useBackgroundIndex);
-		defined.add(completionStyle);
-		defined.add(prettyPrint);
-		defined.add(queryDriver);
-		defined.add(overrideString(additionalOptions, //
+		return overrideOne(defaults, overrideString(additionalOptions, //
```

Comment:
@jonahgraham do you think this code is easy enough for extender?

Reply:
lgtm.